test/ejabberd_tests/tests/ejabberdctl_SUITE.erl

```diff
@@ -595,10 +610,188 @@ stats_host(Config) ->
+        JidA = nick_to_jid(mike, Config),
+        JidB = nick_to_jid(kate, Config),
+        JidRecordAlice = rpc_apply(jlib, binary_to_jid, [JidA]),
+        JidRecordBob = rpc_apply(jlib, binary_to_jid, [JidB]),
+        Msg1 = escalus_stanza:chat_to(<<"kate@localhost">>, "Rolling stones"),
+        Msg2 = escalus_stanza:chat_to(<<"kate@localhost">>, "Arctic monkeys!"),
+        Msg3 = escalus_stanza:chat_to(<<"kate@localhost">>, "More wine..."),
+        Msg4 = escalus_stanza:chat_to(<<"kate@localhost">>, "kings of leon"),
+        OldTimestamp = fallback_timestamp(10, now()),
+        ExpirationTime = fallback_timestamp(2, now()),
+        ExpirationTimeFuture= fallback_timestamp(-5, now()),
+        OfflineOld = generate_offline_expired_message(JidRecordAlice, JidRecordBob, Msg1, OldTimestamp, ExpirationTime),
+        OfflineNow = generate_offline_expired_message(JidRecordAlice, JidRecordBob, Msg2, now(), ExpirationTime),
+        OfflineFuture = generate_offline_expired_message(JidRecordAlice, JidRecordBob, Msg3, now(), ExpirationTimeFuture),
+        OfflineFuture2 = generate_offline_expired_message(JidRecordAlice, JidRecordBob, Msg4, OldTimestamp, ExpirationTimeFuture),
+        {jid, _, _, _, LUser, LServer, _} = JidRecordBob,
+        rpc_apply(mod_offline_backend, write_messages, [LUser, LServer, [OfflineOld, OfflineNow, OfflineFuture, OfflineFuture2], 100]),
+        {selected, _, List} = rpc_apply(ejabberd_odbc, sql_query, [LServer, [<<"Select * from offline_message where username='kate'">>]]),
+        4 = length(List),
+        %% when
+        {_, _} = ejabberdctl("delete_expired_messages", [], Config),
+        {selected, _,  SecondList} = rpc_apply(ejabberd_odbc, sql_query, [LServer, [<<"Select * from offline_message where username='kate'">>]]),
+        %% then
+        2 = length(SecondList)
+    end).
+
 %%-----------------------------------------------------------------
 %% Helpers
 %%-----------------------------------------------------------------
 
+
+nick_to_jid(UserName, Config) when is_atom(UserName) ->
+    UserSpec = escalus_users:get_userspec(Config, UserName),
+    escalus_utils:jid_to_lower(escalus_users:get_jid(Config, UserSpec)).
+
+generate_offline_message(From, To, Msg, TimeStamp) ->
+    {jid, _, _, _, LUser, LServer, _} = To,
+    #offline_msg{us={LUser, LServer}, timestamp=TimeStamp, expire=never,from=From, to=To, packet=Msg}.
+
+generate_offline_expired_message(From, To, Msg, TimeStamp, ExpirationTime) ->
+    {jid, _, _, _, LUser, LServer, _} = To,
+    #offline_msg{us={LUser, LServer}, timestamp=TimeStamp, expire=ExpirationTime,from=From, to=To, packet=Msg}.
+
+
+fallback_timestamp(Days, {MegaSecs, Secs, _MicroSecs}) ->
+    S = MegaSecs * 1000000 + Secs - 60 * 60 * 24 * Days,
+    MegaSecs1 = S div 1000000,
+    Secs1 = S rem 1000000,
+    {MegaSecs1, Secs1, 0}.
+
+encode_timestamp(TimeStamp) ->
+    integer_to_list(now_to_microseconds(TimeStamp)).
+
+now_to_microseconds({Mega, Secs, Micro}) ->
```

Comment:
In tests we have the `usec` library from @pzel that does exactly what you need here :)